crypto/merklekeystore/keystore.go

```diff
@@ -135,22 +145,34 @@ func New(firstValid, lastValid, interval uint64, sigAlgoType crypto.AlgorithmTyp
 		}
 		keys[i] = *sigAlgo
 	}
+
 	s := &Signer{
-		SignatureAlgorithms: keys,
+		keyStore:            PersistentKeystore{store},
+		signatureAlgorithms: keys,
 		FirstValid:          firstValid,
-		ArrayBase:           0,
 		Interval:            interval,
 		mu:                  deadlock.RWMutex{},
```

Comment:
I'm leaving it for now since I can't be sure yet, need to think about it some more. If someone Trims in the middle of a Sign operation some problems may arise, but from a quick glance at the logic I don't really see how that could happen. The first operation in both Trim and Sign is basically an atomic DB operation.